packages/core/src/components/fields/SchemaField.js

```diff
@@ -302,7 +302,9 @@ function SchemaFieldRender(props) {
   if (wasPropertyKeyModified) {
     label = name;
   } else {
-    label = uiSchema["ui:title"] || props.schema.title || schema.title || name;
+    label = schema.hasOwnProperty(ADDITIONAL_PROPERTY_FLAG)
+      ? name
+      : uiSchema["ui:title"] || props.schema.title || schema.title || name;
```

Comment:
I'm not sure quite what this is fixing. Can you add a test -- which should make it clearer the expected behavior?

Reply:
You can see in the demo that the key input field contains the title of the field instead of the key of the addition property. I'll add some tests but I wanted to make sure that it makes sense first

Reply:
Sorry about the late reply. Going to add some tests now, but to clarify the diff:

Before:
<img width="637" alt="image" src="https://user-images.githubusercontent.com/17147717/175841936-d3256b38-d538-48a8-b47a-e051cec8326e.png">

After:
<img width="636" alt="image" src="https://user-images.githubusercontent.com/17147717/175841986-97aee511-bb12-4512-ad7f-95e182f5ee7d.png">

Notice that the key is is `propTitle` which is the title of the field instead of the actual key `fsdf` 

This is the [demo playground](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJmaXJzdE5hbWUiOiJDaHVjayIsImxhc3ROYW1lIjoiTm9ycmlzIiwiaGVsbG8iOnsiZnNkZiI6eyJmaXJzdE5hbWUiOiJoZXkifX19LCJzY2hlbWEiOnsidGl0bGUiOiJBIGN1c3RvbWl6YWJsZSByZWdpc3RyYXRpb24gZm9ybSIsImRlc2NyaXB0aW9uIjoiQSBzaW1wbGUgZm9ybSB3aXRoIGFkZGl0aW9uYWwgcHJvcGVydGllcyBleGFtcGxlLiIsInR5cGUiOiJvYmplY3QiLCJyZXF1aXJlZCI6WyJmaXJzdE5hbWUiLCJsYXN0TmFtZSJdLCJwcm9wZXJ0aWVzIjp7ImZpcnN0TmFtZSI6eyJ0eXBlIjoic3RyaW5nIiwidGl0bGUiOiJGaXJzdCBuYW1lIn0sImxhc3ROYW1lIjp7InR5cGUiOiJzdHJpbmciLCJ0aXRsZSI6Ikxhc3QgbmFtZSJ9LCJoZWxsbyI6eyJ0eXBlIjoib2JqZWN0IiwicHJvcGVydGllcyI6e30sImFkZGl0aW9uYWxQcm9wZXJ0aWVzIjp7IiRyZWYiOiIjLyRkZWZzL2JsYSJ9fX0sIiRkZWZzIjp7ImJsYSI6eyJ0eXBlIjoib2JqZWN0IiwidGl0bGUiOiJwcm9wVGl0bGUiLCJwcm9wZXJ0aWVzIjp7ImZpcnN0TmFtZSI6eyJ0eXBlIjoic3RyaW5nIiwidGl0bGUiOiJGaXJzdCBuYW1lIn0sImxhc3ROYW1lIjp7InR5cGUiOiJzdHJpbmciLCJ0aXRsZSI6Ikxhc3QgbmFtZSJ9fX19fSwidWlTY2hlbWEiOnsiZmlyc3ROYW1lIjp7InVpOmF1dG9mb2N1cyI6dHJ1ZSwidWk6ZW1wdHlWYWx1ZSI6IiIsInVpOmF1dG9jb21wbGV0ZSI6ImZhbWlseS1uYW1lIn0sImxhc3ROYW1lIjp7InVpOmVtcHR5VmFsdWUiOiIiLCJ1aTphdXRvY29tcGxldGUiOiJnaXZlbi1uYW1lIn0sImFnZSI6eyJ1aTp3aWRnZXQiOiJ1cGRvd24iLCJ1aTp0aXRsZSI6IkFnZSBvZiBwZXJzb24iLCJ1aTpkZXNjcmlwdGlvbiI6IihlYXJ0aGlhbiB5ZWFyKSJ9LCJiaW8iOnsidWk6d2lkZ2V0IjoidGV4dGFyZWEifSwicGFzc3dvcmQiOnsidWk6d2lkZ2V0IjoicGFzc3dvcmQiLCJ1aTpoZWxwIjoiSGludDogTWFrZSBpdCBzdHJvbmchIn0sImRhdGUiOnsidWk6d2lkZ2V0IjoiYWx0LWRhdGV0aW1lIn0sInRlbGVwaG9uZSI6eyJ1aTpvcHRpb25zIjp7ImlucHV0VHlwZSI6InRlbCJ9fX0sInRoZW1lIjoiZGVmYXVsdCIsImxpdmVTZXR0aW5ncyI6eyJ2YWxpZGF0ZSI6ZmFsc2UsImRpc2FibGUiOmZhbHNlLCJyZWFkb25seSI6ZmFsc2UsIm9taXRFeHRyYURhdGEiOmZhbHNlLCJsaXZlT21pdCI6ZmFsc2V9fQ==)

Reply:
@epicfaace I finally added the test. Sorry about the delay again